compiler/src/iree/compiler/Codegen/Common/test/tile_and_distribute_workgroups_using_forall.mlir

```diff
@@ -1367,52 +1367,64 @@ attributes {translation_info = #iree_codegen.translation_info<pipeline = LLVMGPU
-  %6 = tensor.empty() : tensor<4x1x96xf16>
-  %7 = tensor.empty() : tensor<4x1xi32>
-  %8 = tensor.empty() : tensor<4x1xf16>
-  %9 = linalg.fill {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[1, 128]]>} ins(%cst : f16) outs(%8 : tensor<4x1xf16>) -> tensor<4x1xf16>
-  %10 = linalg.fill {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[1, 128]]>} ins(%c0_i32 : i32) outs(%7 : tensor<4x1xi32>) -> tensor<4x1xi32>
-  %11:2 = scf.forall (%arg0) = (0) to (128256) step (1336) shared_outs(%arg1 = %6, %arg2 = %5) -> (tensor<4x1x96xf16>, tensor<4x1x96xi32>) {
-    %12 = arith.cmpi slt, %arg0, %c0 : index
-    %13 = arith.subi %c-1, %arg0 : index
-    %14 = arith.select %12, %13, %arg0 : index
-    %15 = arith.divsi %14, %c1336 : index
-    %16 = arith.subi %c-1, %15 : index
-    %17 = arith.select %12, %16, %15 : index
-    %18 = iree_tensor_ext.dispatch.tensor.load %0, offsets = [0, 0, %arg0], sizes = [4, 1, 1336], strides = [1, 1, 1] : !iree_tensor_ext.dispatch.tensor<readonly:tensor<4x1x128256xf16>> -> tensor<4x1x1336xf16>
-    %extracted_slice = tensor.extract_slice %arg1[0, 0, %17] [4, 1, 1] [1, 1, 1] : tensor<4x1x96xf16> to tensor<4x1x1xf16>
-    %broadcasted = linalg.broadcast ins(%9 : tensor<4x1xf16>) outs(%extracted_slice : tensor<4x1x1xf16>) dimensions = [2]  {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[1, 128]]>}
-    %extracted_slice_0 = tensor.extract_slice %broadcasted[0, 0, 0] [4, 1, 1] [1, 1, 1] : tensor<4x1x1xf16> to tensor<4x1xf16>
-    %extracted_slice_1 = tensor.extract_slice %arg2[0, 0, %17] [4, 1, 1] [1, 1, 1] : tensor<4x1x96xi32> to tensor<4x1x1xi32>
-    %broadcasted_2 = linalg.broadcast ins(%10 : tensor<4x1xi32>) outs(%extracted_slice_1 : tensor<4x1x1xi32>) dimensions = [2]  {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[1, 128]]>}
-    %extracted_slice_3 = tensor.extract_slice %broadcasted_2[0, 0, 0] [4, 1, 1] [1, 1, 1] : tensor<4x1x1xi32> to tensor<4x1xi32>
-    %19 = arith.muli %17, %c1336 : index
-    %20:2 = iree_linalg_ext.arg_compare {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[1, 128]]>} dimension(2) ins(%18 : tensor<4x1x1336xf16>) outs(%extracted_slice_0, %extracted_slice_3 : tensor<4x1xf16>, tensor<4x1xi32>) index_base(%19 : index) {
-    ^bb0(%arg3: f16, %arg4: f16):
-      %21 = arith.cmpf ogt, %arg3, %arg4 : f16
-      iree_linalg_ext.yield %21 : i1
+  %empty_f16 = tensor.empty() : tensor<4x1x96xf16>
+  %empty_i32 = tensor.empty() : tensor<4x1x96xi32>
+  %init_f16 = tensor.empty() : tensor<4x1xf16>
+  %init_i32 = tensor.empty() : tensor<4x1xi32>
+  %fill_f16 = linalg.fill {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[1, 128]]>}
+      ins(%cst : f16) outs(%init_f16 : tensor<4x1xf16>) -> tensor<4x1xf16>
+  %fill_i32 = linalg.fill {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[1, 128]]>}
+      ins(%c0_i32 : i32) outs(%init_i32 : tensor<4x1xi32>) -> tensor<4x1xi32>
```

Comment:
Could these also be function arguments? Do we care about the lowering config (I don't know)